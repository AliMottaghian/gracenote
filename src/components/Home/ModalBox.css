.modalBox {
  padding: 10px;
  border: 1px solid var(--dark-gray);
  margin: 1em 10px;
}
.modalBox.inline{
  border:none;
  display: flex;
}


