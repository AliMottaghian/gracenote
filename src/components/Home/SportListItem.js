import React from "react";
import PropTypes from "prop-types";
import { Row, Col, Button } from "antd";
import { UnorderedListOutlined } from "@ant-design/icons";

const SportListItem = ({ history, name, id }) => {
  const showDetails = () => {
    history.push(`/detail/${id}`);
  };
  return (
    <Row>
      <Col xs={18} md={20}>
        <strong style={{ lineHeight: 2 }}>{name}</strong>
      </Col>
      <Col xs={6} md={4}>
        <Button
          type="default"
          icon={<UnorderedListOutlined />}
          onClick={showDetails}
        >
          Details
        </Button>
      </Col>
    </Row>
  );
};
SportListItem.propTypes = {
  history: PropTypes.object.isRequired,
  name: PropTypes.string,
  id: PropTypes.number.isRequired,
};

export default SportListItem;
