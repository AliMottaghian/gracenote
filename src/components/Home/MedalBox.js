import React from "react";
import PropTypes from "prop-types";
import { Col } from "antd";
import "./ModalBox.css";

const MedalBox = ({ type, value }) => {
  return (
    <Col xs={12} md={6} className="textCenter">
      <div className={`modalBox`}>
        <h3>
          <span className={`medalIndicator ${type}`}></span>
          {type.split("_")[1]}
        </h3>
        <b>{value}</b>
      </div>
    </Col>
  );
};

MedalBox.propTypes = {
  type: PropTypes.string.isRequired,
  value:PropTypes.number
};
export default MedalBox;
