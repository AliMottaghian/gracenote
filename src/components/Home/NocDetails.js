import React from "react";
import PropTypes from "prop-types";
import { Row } from "antd";

import MedalBox from "./MedalBox";

const NocDetails = ({ medals, country }) => {
  return (
    <div>
      <h3>{country} - Medals</h3>

      <Row>
        {Object.keys(medals).map((medal) => {
          return (
            <MedalBox key={Math.random()} type={medal} value={medals[medal]} />
          );
        })}
      </Row>
    </div>
  );
};
NocDetails.propTypes = {
  medals: PropTypes.object.isRequired,
  country: PropTypes.string.isRequired,
};
export default NocDetails;
