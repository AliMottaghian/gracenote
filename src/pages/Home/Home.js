import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { getNocDetail, getSportsList } from "../../utils/dataProvider";
import Card from "../../components/Ui/Card/Card";
import NocDetails from "../../components/Home/NocDetails";
import SportListItem from "../../components/Home/SportListItem";

const Home = ({ history }) => {
  const [nocMedals, setNocMedals] = useState(null);
  const [sportsList, setSportsList] = useState(null);

  useEffect(() => {
    const nocDetail = getNocDetail();
    const sportsList = getSportsList();
    setNocMedals(nocDetail);
    setSportsList(sportsList);
  }, []);
  return (
    <>
      <div>
        <Card>
          {nocMedals?.Medals && (
            <NocDetails
              medals={nocMedals.Medals}
              country={`${nocMedals?.NOC.c_Name} (${nocMedals?.NOC.c_Short})`}
            />
          )}
        </Card>
      </div>
      {sportsList?.map((item) => {
        return (
          <Card key={item.Sport.n_ID}>
            <SportListItem
              history={history}
              name={item.Sport.c_Name}
              id={item.Sport.n_ID}
            />
          </Card>
        );
      })}
    </>
  );
};

Home.propTypes = {
  history: PropTypes.object
};
export default Home;
