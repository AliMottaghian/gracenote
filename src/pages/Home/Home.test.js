import Home from "./Home";

describe('<Home />',()=>{
  const wrapper =  shallow(<Home />);
  it("should render App component without crashing",()=>{
    expect(wrapper).toMatchSnapshot()
  })
})