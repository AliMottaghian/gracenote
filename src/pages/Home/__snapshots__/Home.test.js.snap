// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render App component without crashing 1`] = `
<Fragment>
  <div>
    <Card />
  </div>
</Fragment>
`;
